### PR TITLE
Evaporation rate scales with timestep length

### PR DESCRIPTION
**Pull request by LaurelineJ**

### CODE REVIEW

**Comment by jrising** [comment]
It would be great to merge this before the holidays.  Is it ready, as soon as you fix the checks?

File: src/Reservoir.jl [hunk 1]
```diff
@@ -4,15 +4,23 @@
 using Mimi
 using Distributions
 
-reservoirdata=readtable(datapath("reservoirs/allreservoirs.csv"))
+if config["netset"] == "usa"
+	reservoirdata=readtable(datapath("reservoirs/allreservoirs.csv"))
+end
```
Comment (jrising) [inline src/Reservoir.jl]: What happens if netset <> "usa"?  Won't it just fail later?

File: src/Reservoir.jl [hunk 2]
```diff
@@ -63,30 +92,27 @@ function initreservoir(m::Model, name=nothing)
...
+	   rcmax = convert(Vector{Float64}, reservoirdata[:MAXCAP])
      	   rcmax = rcmax*1233.48
      	   reservoir[:storagecapacitymax] = rcmax;
-     	   reservoir[:storagecapacitymin] = 0*0.1*rcmax;
-     	   reservoir[:storage0] = 0*(rcmax-0.1*rcmax)/2; #initial storate value: (max-min)/2
-     	   reservoir[:evaporation] = 0.01*ones(m.indices_counts[:reservoirs],m.indices_counts[:time]);
-     	   reservoir[:captures] = cached_fallback("extraction/captures", () -> zeros(m.indices_counts[:reservoirs], m.indices_counts[:time]));
-	   end
+	   reservoir[:storagecapacitymin] = zeros(numreservoirs);
+	   reservoir[:storage0] = rcmax*0.#zeros(numreservoirs);
```
Comment (jrising) [inline src/Reservoir.jl]: Is this just to avoid giving regions free water?  It's a pretty weird assumption that all reservoirs start empty.

Comment (LaurelineJ) [inline src/Reservoir.jl]: Yes. I believe your initial idea was to see the storage not as an absolute value but the amount that we can actually use (thus we do not have to worry about minimum amount necessary to work).

I think this makes sense, in particular in two set-ups:
1. To explore the role of storage to mitigate seasonality effect (yearly time horizon, monthly timesteps)
2. To explore the role of storage to mitigate persistent droughts (10 years time horizon)

**Comment by LaurelineJ** [comment]
@jrising I still have doubts about what I did in Reservoir.jl line 47-54, but other than this, I think this can be merged.

File: src/Reservoir.jl [hunk 3]
```diff
@@ -32,27 +40,50 @@ function run_timestep(c::Reservoir, tt::Int)
...
+
+    v.inflows[:,tt] = zeros(numreservoirs); 
+    v.outflows[:,tt] = zeros(numreservoirs); 
+    rr = 1
+    for gg in 1:numgauges
+       index = vertex_index(downstreamorder[gg])
+       if isreservoir[index] > 0
+	     v.inflows[rr,tt] = p.inflowsgauges[gg, tt]*1000; 
+	     v.outflows[rr,tt] = p.outflowsgauges[gg, tt]*1000; 
+	     rr += 1
```
Comment (jrising) [inline src/Reservoir.jl]: This relies on reservoirs being in the same order as the downstream order of gauges, but I don't see why that would be the case.  However, `isreservoir[index]` already gives the reservoir index for a give node.  So you should just be able to say:
```
rr = isreservoir[index]
v.inflows[rr,tt] = p.inflowsgauges[gg, tt]*1000;
v.outflows[rr,tt] = p.outflowsgauges[gg, tt]*1000; 
```
or
```
v.inflows[isreservoir[index],tt] = p.inflowsgauges[gg, tt]*1000; 
v.outflows[isreservoir[index],tt] = p.outflowsgauges[gg, tt]*1000;
```

Comment (LaurelineJ) [inline src/Reservoir.jl]: Fabulous ! I am always a bit confused when it comes to gauges-reservoirs-counties connections ... I'll make the changes

Comment (jrising) [reply src/Reservoir.jl]: I think if you make this change, we can merge it in.

**Comment by LaurelineJ** [comment]
I think it is ok now ...

File: src/lib/datastore.jl [hunk 4]
```diff
@@ -52,9 +52,9 @@ function cached_fallback(filename, generate)
     confighash = hash(config) # make specific to configs
     try
         if isfile(datapath("$filename$suffix-$confighash.jld"))
-            return deserialize(open(cachepath("$filename$suffix-$confighash.jld")))
+            return deserialize(open(datapath("$filename$suffix-$confighash.jld")))
```
Comment (LaurelineJ) [inline src/lib/datastore.jl]: maybe better to revert these changes and change line 54 to cachepath

Comment (jrising) [reply src/lib/datastore.jl]: Agreed.  And I think you can simplify this whole section, since a lot of the logic is now in cache path.

**Comment by jrising** [comment]
Awesome!  Want to merge it?
